src/AutoRest.CSharp/Common/Output/Models/ValueExpressions.cs

```diff
@@ -0,0 +1,291 @@
+
+                public static ValueExpression CreatePageable(
+                    CodeWriterDeclaration createFirstPageRequest,
+                    CodeWriterDeclaration? createNextPageRequest,
+                    ValueExpression clientDiagnostics,
+                    ValueExpression pipeline,
+                    CSharpType? pageItemType,
+                    string scopeName,
+                    string itemPropertyName,
+                    string? nextLinkPropertyName,
+                    ValueExpression? requestContextOrCancellationToken,
+                    bool async)
+                {
+                    var arguments = new List<ValueExpression>
+                    {
+                        new VariableReference(createFirstPageRequest),
+                        createNextPageRequest != null ? new VariableReference(createNextPageRequest) : Null,
+                        GetValueFactory(pageItemType),
+                        clientDiagnostics,
+                        pipeline,
+                        Literal(scopeName),
+                        Literal(itemPropertyName),
+                        Literal(nextLinkPropertyName)
+                    };
+
+                    if (requestContextOrCancellationToken is not null)
+                    {
+                        arguments.Add(requestContextOrCancellationToken);
+                    }
+
+                    var methodName = async ? nameof(Azure.Core.PageableHelpers.CreateAsyncPageable) : nameof(Azure.Core.PageableHelpers.CreatePageable);
+                    return new StaticMethodCallExpression(PageableHelpersType, methodName, arguments, false, false);
+                }
+
+                public static ValueExpression CreatePageable(
+                    CodeWriterDeclaration message,
+                    CodeWriterDeclaration? createNextPageRequest,
+                    CodeWriterDeclaration clientDiagnostics,
+                    CodeWriterDeclaration pipeline,
+                    CSharpType? pageItemType,
+                    OperationFinalStateVia finalStateVia,
+                    string scopeName,
+                    string itemPropertyName,
+                    string? nextLinkPropertyName,
+                    ValueExpression? requestContext,
+                    bool async)
+                {
+                    var arguments = new List<ValueExpression>
+                    {
+                        new ParameterReference(KnownParameters.WaitForCompletion),
+                        new VariableReference(message),
+                        createNextPageRequest is not null ? new VariableReference(createNextPageRequest) : Null,
+                        GetValueFactory(pageItemType),
+                        new VariableReference(clientDiagnostics),
+                        new VariableReference(pipeline),
+                        new FormattableStringToExpression($"{typeof(OperationFinalStateVia)}.{finalStateVia}"),
```

Comment:
Should we create a method like `PropertyReference` or `PropertyInvocation` to handle case like this?
like 
```
public static ValueExpression PropertyReference(CSharpType type, string propertyName)
```

Reply:
Here it is not a property reference, but rather an enum value. A snippet is created to handle this and similar cases